pkg/backupdriver/backup_driver_controller.go

```diff
@@ -289,7 +289,7 @@ func (ctrl *backupDriverController) cloneFromSnapshot(cloneFromSnapshot *backupd
 		}
 		return err
 	}
-	if pvc.Spec.StorageClassName != nil {
+	if pvc.Spec.StorageClassName != nil && (*pvc.Spec.StorageClassName) != ""{
```

Comment:
pvc.Spec.StorageClassName is not a pointer.  Add a check for `pvc.Spec.StorageClassName) != ""` should be good.

Reply:
pvc.Spec.StorageClassName is a string pointer https://github.com/kubernetes/api/blob/master/core/v1/types.go#L489

pv.spec.storageClassName is a plain string https://github.com/kubernetes/api/blob/master/core/v1/types.go#L351

Reply:
You are right.  I was looking at pv spec.

It looks good.